src/System/include/BipedalLocomotion/System/FloatingBaseDynamicalSystem.h

```diff
@@ -0,0 +1,116 @@
+ * The FloatingBaseDynamicalSystem inherits from a generic DynamicalSystem where:
+ * - DynamicalSystem::StateType is described by an std::tuple containing:
+ *   - iDynTree::Vector6: the base velocity expressed in mixed representation;
+ *   - iDynTree::VectorDynsize: the joint velocities [in rad/s];
+ *   - iDynTree::Position: position of the base w.r.t. the inertial frame
+ *   - iDynTree::Rotation: rotation matrix \f${} ^ I R _ {b}\f$. Matrix that transform a vector
+ * whose coordinates are expressed in the base frame in the inertial frame;
+ *   - iDynTree::VectorDynsize: the joint positions [in rad].
+ * - DynamicalSystem::StateDerivativeType is described by an std::tuple containing:
+ *   - iDynTree::Vector6: the base acceleration expressed in mixed representation;
+ *   - iDynTree::VectorDynsize: the joint accelerations [in rad/s^2];
+ *   - iDynTree::Vector3: base velocity w.r.t. the inertial frame;
+ *   - iDynTree::Matrix3x3: rate of change of the rotation matrix \f${} ^ I \dot{R} _ {b}\f$.
+ * whose coordinates are expressed in the base frame in the inertial frame;
+ *   - iDynTree::VectorDynsize: the joint velocities [in rad/s].
+ * - DynamicalSystem::InputType is described by an std::tuple containing:
+ *   - iDynTree::VectorDynsize: the joint torques [in Nm];
+ *   - std::vector<ContactWrench>: List of contact wrenches.
+ */
+class FloatingBaseDynamicalSystem : public DynamicalSystem<std::tuple<iDynTree::Vector6,
+                                                                      iDynTree::VectorDynSize,
+                                                                      iDynTree::Position,
+                                                                      iDynTree::Rotation,
+                                                                      iDynTree::VectorDynSize>,
+                                                           std::tuple<iDynTree::Vector6,
+                                                                      iDynTree::VectorDynSize,
+                                                                      iDynTree::Vector3,
+                                                                      iDynTree::Matrix3x3,
+                                                                      iDynTree::VectorDynSize>,
+                                                           std::tuple<iDynTree::VectorDynSize,
+                                                                      std::vector<ContactWrench>>>
+{
+    static constexpr size_t m_baseDoFs = 6; /**< Number of degree of freedom associated to the
+                                               floating base */
+
+    std::shared_ptr<iDynTree::KinDynComputations> m_kinDyn; /**< Pointer to an existing instance of
+                                                               kinDynComputations object */
+    std::size_t m_actuatedDoFs{0}; /**< Number of actuated degree of freedom */
+
+    iDynTree::Vector3 m_gravity; /**< Gravity vector */
```

Comment:
Personal experience: offer a way of setting this, as the hardcoded -9.81 will bite you eventually .